llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -505,11 +505,14 @@ void OpenMPIRBuilder::getKernelArgsVector(TargetKernelArgs &KernelArgs,
 
   Value *NumTeams3D =
       Builder.CreateInsertValue(ZeroArray, KernelArgs.NumTeams[0], {0});
+  Value *NumThreads3D =
+      Builder.CreateInsertValue(ZeroArray, KernelArgs.NumThreads[0], {0});
   for (unsigned I = 1; I < std::min(KernelArgs.NumTeams.size(), MaxDim); ++I)
     NumTeams3D =
         Builder.CreateInsertValue(NumTeams3D, KernelArgs.NumTeams[I], {I});
-  Value *NumThreads3D =
-      Builder.CreateInsertValue(ZeroArray, KernelArgs.NumThreads, {0});
+  for (unsigned I = 1; I < std::min(KernelArgs.NumThreads.size(), MaxDim); ++I)
```

Comment:
`for (unsigned I : seq<unsigned>(1, std::min(KernelArgs.NumThreads.size(), MaxDim)))`